index.js

```diff
@@ -1,35 +1,40 @@
 /**
  * PCM volume controller.
- * Same as GainNode in web-audio-api, but for streams.
+ * Same as GainNode in web-audio-api as a generic function.
+ * stream in stream.js, pull-stream in pull.js, element in element.js
  *
  * @module audio-gain
  */
-
+var noop = function () {};
 var util = require('audio-buffer-utils');
 
 /**
- * Create pcm stream volume controller.
+ * Create pcm volume controller.
  *
  * @function
  */
 function gain(options) {
-	if (typeof options === 'number') {
+	if (!options) options = 1;
+	if (typeof options === 'number' || typeof options.volume === 'number') {
+		// adjust gain(Number) || gain({ volume: Number })
+		var volume = options.volume || options;
+		options = {volume: function (x) { return x * volume }};
+	} else if (typeof options === 'function')  {
+		// adjust gain(Function);
 		options = {volume: options};
```

Comment:
With this solution force user to decide how he should calculate volume - multiplication, tangent or whatever. He should do calculation and return a new sample value. In this case indeed there is no difference between _audio-processor_.
Instead I would suggest asking user for a new _volume_ by calling a function, but making multiplication ourselves.